### PR TITLE
fix: Enable lib checks

### PR DESCRIPTION
**Pull request by CodeSchwert**
# Summary

[DX-2487](https://immutable.atlassian.net/browse/DX-2487)

Make `skipLibCheck` `false` in the SDK `tsconfig.json`. Some packages were generating incorrect / broken type declarations. These kinds of errors will now be detected by typecheck.

# Customer Impact

Fewer broken type declarations in the package.

## Fixed

# Things worth calling out

# Before submitting the PR, please consider the following:

- [ ] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.

[DX-2487]: https://immutable.atlassian.net/browse/DX-2487?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +37,8 @@
   "private": true,
   "repository": "immutable/ts-immutable-sdk.git",
   "resolutions": {
-    "@openzeppelin/contracts": "3.4.2-solc-0.7"
+    "@openzeppelin/contracts": "3.4.2-solc-0.7",
+    "responselike": "^2.0.0"
```
Comment (keithbro-imx) [inline package.json:41]: Seems like this should have a comment explaining why its needed.

File: package.json [hunk 2]
```diff
@@ -37,7 +37,8 @@
   "private": true,
   "repository": "immutable/ts-immutable-sdk.git",
   "resolutions": {
-    "@openzeppelin/contracts": "3.4.2-solc-0.7"
+    "@openzeppelin/contracts": "3.4.2-solc-0.7",
```
Comment (keithbro-imx) [inline package.json:40]: Should we remove this ?

**Review by keithbro-imx — APPROVED** [review]

**Review by allan-almeida-imtbl — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (CodeSchwert) [reply package.json:40]: I think we should be able to remove this.

File: package.json (hunk 1 above)
Comment (CodeSchwert) [reply package.json:41]: The `responseLike` resolution is to fix a broken type in one Yarn's dependencies. Upgrading won't fix it, as the latest version of Yarn would still include the broken package, which is used in `cacheable-request`.

**Review by keithbro-imx — DISMISSED** [review]

**Review by allan-almeida-imtbl — DISMISSED** [review]

**Review by proletesseract — DISMISSED** [review]

**Review by imx-mikhala — APPROVED** [review]

**Review by carleeto — APPROVED** [review]

**Review by proletesseract — APPROVED** [review]

**Review by valzam — APPROVED** [review]

**Review by scastleman-immutable — APPROVED** [review]

**Review by keithbro-imx — APPROVED** [review]

**Pull request merged**